packages/next-on-pages/src/buildApplication/buildMetadataFiles.ts

```diff
@@ -40,22 +40,37 @@ async function buildNextStaticHeaders(
 	);
 	const nextStaticHeaders = (nextStaticRoute as VercelSource)?.headers;
 
-	if (nextStaticHeaders) {
-		await writeFile(
-			join(outputDir, '_headers'),
-			`
+	if (!nextStaticHeaders) {
+		return;
+	}
+
+	const headersPath = join(outputDir, '_headers');
+
+	const nopContent = `
 # === START AUTOGENERATED @cloudflare/next-on-pages IMMUTABLE HEADERS ===
 ${nextStaticPath}/*
 ${Object.entries(nextStaticHeaders)
 	.map(([header, value]) => `  ${header}: ${value}`)
 	.join('\n')}
-# === END AUTOGENERATED @cloudflare/next-on-pages IMMUTABLE HEADERS ===\n`,
-			{
-				// in case someone configured redirects already, append to the end
-				flag: 'a',
-			},
-		);
+# === END AUTOGENERATED @cloudflare/next-on-pages IMMUTABLE HEADERS ===\n`;
+
+	try {
+		// Vercel hard links the static files directory:
+		// https://github.com/vercel/vercel/blob/8e20fed2769ff0169179d2d40fcc97cfd07a8bfe/packages/cli/src/util/build/write-build-result.ts#L336
```

Comment:
we usually shorten the sha not to make the line too long 🙂 
```suggestion
		// https://github.com/vercel/vercel/blob/8e20fed/packages/cli/src/util/build/write-build-result.ts#L336
```